### PR TITLE
Clippy Fixes

### PR DESCRIPTION
**Pull request by AsceticBear**
Fix https://github.com/darwinia-network/darwinia-common/issues/1220

For those heavily bounded with upstream repo, use `#![allow(clippy::all)]`, such as balances, staking, ethereum, evm, transaction_pause.

TODO: Add Clippy check to CI

### CODE REVIEW

**Comment by AurevoirXavier** [comment]
It's okay. But low priority.

File: frame/bridge/relayer-game/src/lib.rs [hunk 1]
```diff
@@ -575,9 +576,8 @@ impl<T: Config<I>, I: Instance> Module<T, I> {
 
 							if previous_maybe_extended_relay_affirmation_id.is_none() {
 								let relay_header_parcels = &round_relay_affirmations
-									.into_iter()
-									.nth(index as usize)
-									.unwrap()
+									.get(index as usize)
+									.unwrap() // TODO: Remove this unwrap
```
Comment (AsceticBear) [inline frame/bridge/relayer-game/src/lib.rs:580]: @AurevoirXavier Do we need to deal with this?

Comment (AurevoirXavier) [reply frame/bridge/relayer-game/src/lib.rs:580]: relayer game will be removed while eth2.0 released.

File: node/runtime/pangolin/src/lib.rs [hunk 2]
```diff
@@ -123,7 +124,7 @@ pub const VERSION: RuntimeVersion = RuntimeVersion {
 	spec_name: sp_runtime::create_runtime_str!("Pangolin"),
 	impl_name: sp_runtime::create_runtime_str!("Pangolin"),
 	authoring_version: 0,
-	spec_version: 2_9_02_0,
+	spec_version: 29_020,
```
Comment (HackFisher) [inline node/runtime/pangolin/src/lib.rs:127]: @AurevoirXavier Look good for you?

File: node/runtime/pangoro/src/lib.rs [hunk 3]
```diff
@@ -93,7 +94,7 @@ pub const VERSION: RuntimeVersion = RuntimeVersion {
 	spec_name: sp_runtime::create_runtime_str!("Pangoro"),
 	impl_name: sp_runtime::create_runtime_str!("Pangoro"),
 	authoring_version: 0,
-	spec_version: 2_9_02_0,
+	spec_version: 29_020,
```
Comment (HackFisher) [inline node/runtime/pangoro/src/lib.rs:97]: Same

File: node/runtime/pangolin/src/lib.rs (hunk 2 above)
Comment (AurevoirXavier) [reply node/runtime/pangolin/src/lib.rs:127]: Please keep the origin style here. @AsceticBear

Comment (AsceticBear) [reply node/runtime/pangolin/src/lib.rs]: I am okay with both. Fixed

**Review by HackFisher — APPROVED** [review]

**Comment by HackFisher** [comment]
@AsceticBear Please resolve the conflicts

**Comment by AurevoirXavier** [comment]
image

**Pull request merged**